File: LoraMeasurementsBackend/Model/Device.cs
using Newtonsoft.Json;

namespace LoraMeasurementsBackend.Model;

public class Device
{
    public string Id { get; set; }
    public string ApplicationId { get; set; }
    public Location Location { get; set; }
    [JsonIgnore]
    public List<Measurement> Measurements { get; set; }

    public void TryUpdate(string latitude, string longitude, string applicationId,
        double temperature, double barometricPressure, double relativeHumidity, double analogIn,
        DateTime measurementTime, double rssi, double snr, int channelId, double channelRssi)
    {
        Location.Latitude = latitude;
        Location.Longitude = longitude;
        ApplicationId = applicationId;
        Measurements.Add(new Measurement()
        {
            DeviceId = Id,
            Device = this,
            Temperature = temperature,
            BarometricPressure = barometricPressure,
            RelativeHumidity = relativeHumidity,
            AnalogIn = analogIn,
            MeasurementTime = measurementTime,
            Rssi = rssi,
            Snr = snr,
            ChannelId = channelId,
            ChannelRssi = channelRssi,
        });
    }
}

File: LoraMeasurements/src/model/device.ts
export class Device {
    id: string;
    applicationId: string;
    location: Location;


    constructor(id: string, applicationId: string, location: Location) {
        this.id = id;
        this.applicationId = applicationId;
        this.location = location;

    }
}

class Location {
    latitude: string;
    longitude: string;

    constructor(latitude: string, longitude: string) {
        this.latitude = latitude;
        this.longitude = longitude;
    }
}